test/GPv2SafeERC20.test.ts

```diff
@@ -159,7 +175,28 @@ describe("GPv2SafeERC20.sol", () => {
             recipient.address,
             amount,
           ),
-        ).to.be.revertedWith("GPv2SafeERC20: failed transfer");
+        ).to.be.revertedWith("failed transferFrom");
+      });
+
+      it("reverts when invalid ABI encoded bool is returned", async () => {
+        const amount = ethers.utils.parseEther("1.0");
+
+        const sellToken = await waffle.deployMockContract(
+          deployer,
+          ERC20_RETURNING_UINT,
+        );
+        await sellToken.mock.transfer
+          .withArgs(recipient.address, amount)
+          .returns(42);
+
+        await expect(
+          executor.transferFrom(
+            sellToken.address,
+            traders[0].address,
+            recipient.address,
+            amount,
+          ),
+        ).to.be.reverted;
```

Comment:
This is a very sneaky typo with important consequences on the test. Is there a way to require at the end of the test that the mock was called?

```suggestion
        await sellToken.mock.transfer
          .withArgs(recipient.address, amount)
          .returns(42);

        await expect(
          executor.transfer(
            sellToken.address,
            traders[0].address,
            recipient.address,
            amount,
          ),
        ).to.be.reverted;
```

Reply:
Nice catch!

> Is there a way to require at the end of the test that the mock was called?

With Waffle yes, but its currently not supported with Hardhat :cry:

Reply:
Now that I added revert messages for invalid boolean values, this typo was caught since it reverted with "mock not initialized" instead of the expected error :tada:.